.pre-commit-config.yaml

```diff
@@ -1,12 +1,69 @@
+# Learn more about this config here: https://pre-commit.com/
+
+# To enable these pre-commit hooks run:
+# `brew install pre-commit` or `python3 -m pip install pre-commit`
+# Then in the project root directory run `pre-commit install`
+
 default_stages: [commit]
 repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.2.0
+    hooks:
+      - id: check-builtin-literals
+      - id: check-case-conflict
+      - id: check-docstring-first
+      # - id: check-executables-have-shebangs  # TODO: Enable and fix or delete
+      # - id: check-json  # TODO: Enable and fix or delete
+      - id: check-shebang-scripts-are-executable
+      # - id: check-symlinks  # TODO: Enable and fix or delete
+      - id: check-toml
+      - id: check-xml
+      - id: check-yaml
+      # - id: detect-aws-credentials  # TODO: Enable and fix or delete
+      - id: detect-private-key
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+      - id: trailing-whitespace
+
+  - repo: https://github.com/PyCQA/bandit
+    rev: 1.7.4
+    hooks:
+      - id: bandit
+        args:
+          - --skip=B201
+
   - repo: https://github.com/psf/black
     rev: 22.3.0
     hooks:
       -   id: black
+
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.1.0
+    hooks:
+      - id: codespell  # See 'setup.cfg' for args
+
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 3.9.2
+    hooks:
+      - id: flake8
+        additional_dependencies: [flake8-bugbear, flake8-comprehensions]
+        args:
+          - --builtins=Element,PyItemTemplate,PyListTemplate,pyscript
+          - --max-complexity=10
+          - --max-line-length=88
```

Comment:
Personally, while I'm a fan of `black` using 88 chars, I prefer to bump this up to 100, just for the exact case of URLs in docstrings.

Reply:
Fixed.